plugins/tiddlywiki/demo-alternate-store/rawmarkup.js

```diff
@@ -0,0 +1,426 @@
+/*\
+title: $:/plugins/tiddlywiki/demo-alternate-store/rawmarkup.js
+type: application/javascript
+module-type: library
+
+Startup code injected as raw markup
+
+\*/
+
+(function() {
+
+// Need to initialise these because we run before bootprefix.js and boot.js
+$tw = window.$tw || Object.create(null);
+$tw.hooks = $tw.hooks || { names: {}};
+$tw.boot = $tw.boot || {};
+$tw.boot.preloadDirty = $tw.boot.preloadDirty || [];
+
+$tw.Wiki = function(options) {
+    options = options || {};
+    var self = this,
+    tiddlers = Object.create(null), // Hashmap of tiddlers
+    tiddlerTitles = null, // Array of tiddler titles
+    getTiddlerTitles = function() {
+        if(!tiddlerTitles) {
+            tiddlerTitles = Object.keys(tiddlers).sort(function(a,b) {return a.localeCompare(b);});
+        }
+        return tiddlerTitles;
+    },
+    pluginTiddlers = [], // Array of tiddlers containing registered plugins, ordered by priority
+    pluginInfo = Object.create(null), // Hashmap of parsed plugin content
+    shadowTiddlers = Object.create(null), // Hashmap by title of {source:, tiddler:}
+    shadowTiddlerTitles = null,
+    getShadowTiddlerTitles = function() {
+        if(!shadowTiddlerTitles) {
+            shadowTiddlerTitles = Object.keys(shadowTiddlers);
+        }
+        return shadowTiddlerTitles;
+    },
+    enableIndexers = options.enableIndexers || null,
+    indexers = [],
+    indexersByName = Object.create(null);
+    //$tw.utils replacements
+    var eachObj = function(object,callback) {
+        var next,f,length;
+        if(object) {
+            if(Object.prototype.toString.call(object) == "[object Array]") {
+                for (f=0, length=object.length; f<length; f++) {
+                    next = callback(object[f],f,object);
+                    if(next === false) {
+                        break;
+                    }
+                }
+            } else {
+                var keys = Object.keys(object);
+                for (f=0, length=keys.length; f<length; f++) {
+                    var key = keys[f];
+                    next = callback(object[key],key,object);
+                    if(next === false) {
+                        break;
+                    }
+                }
+            }
+        }
+    },
+    hop = function(object,property) {
```

Comment:
Are we redefining too many things? I thought we just need to define some **hooks** to do "[hydration](https://react.dev/reference/react-dom/client/hydrateRoot)"